### PR TITLE
Add `dask config set`

### LINKED ISSUE #10917: Deprecation warning is pretty intense
```python
In [1]: import dask

In [2]: import dask.dataframe
<ipython-input-2-1d704cbd4e0a>:1: DeprecationWarning: The current Dask DataFrame implementation is deprecated.
In a future release, Dask DataFrame will use new implementation that
contains several improvements including a logical query planning.
The user-facing DataFrame API will remain unchanged.

The new implementation is already available and can be enabled by
installing the dask-expr library:

    $ pip install dask-expr

and turning the query planning option on:

    >>> import dask
    >>> dask.config.set({'dataframe.query-planning': True})
    >>> import dask.dataframe as dd

API documentation for the new implementation is available at
https://docs.dask.org/en/stable/dask-expr-api.html

Any feedback can be reported on the Dask issue tracker
https://github.com/dask/dask/issues

  import dask.dataframe

```

Also this happens every time I import `dask.dataframe`, rather than just once.  Is there a good way to turn it off?  This strikes me as scary.

In my mind if we're giving people a scary warning then there needs to be an easy way to opt-out of that warning, either by opting in to the new behavior (if we think that that's wise) or by having an opt-out option for future use

```
To disable this warning in the future, run the following command:

dask config set dataframe.query-optimization-warning=False
```

### PR DESCRIPTION
**Pull request by milesgranger**
- [x] Closes #10921 
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

---

- CLI to set config values. ie. `dask config set dataframe.query-planning-warning False`.

### CODE REVIEW

File: dask/dask.yaml [hunk 1]
```diff
@@ -16,6 +16,7 @@ dataframe:
     metadata-task-size-remote: 1  # Number of files per remote metadata-processing task
   convert-string: null  # Whether to convert string-like data to pyarrow strings
   query-planning: false  # Whether to use dask-expr
+  query-optimization-warning: true  # Warn about future query optimization change
```
Comment (crusaderky) [inline dask/dask.yaml:19]: Could you rename to `query-planning-warning`?

**Review by crusaderky — CHANGES_REQUESTED** [review]
The new command dumps out your _whole_ dask config to `~/.config/dask/dask.yaml`.

I would like to treat this as an antipattern, as it will cause many problems down the line when some developer changes a default of some obscure setting but the users don't pick up the change. What's worse, wiping and rebuilding the conda environment won't reflect the change. Also, the user will have no indication, when they do open that config file, about which settings were their own intentional changes from a month before and which were just copy-pasted defaults.

Could you change the code to add the new explicit config override to whatever was already there in `~/.config/dask/dask.yaml`, but nothing in addition to that?

Finally, could you document the new command in `docs/source/configuration.rst`?

File: dask/tests/test_cli.py [hunk 2]
```diff
@@ -34,6 +35,35 @@ def test_config_get_bad_value():
...
+        ["array.chunk-size", "333MiB"],
+        env={"DASK_CONFIG": str(tmp_conf_dir)},
+    )
+
+    assert config_file.exists()
+
+    expected = (
+        f"Updated [array.chunk-size] to [333MiB], config saved to {config_file}\n"
+    )
+    assert expected == result.output
```
Comment (crusaderky) [inline dask/tests/test_cli.py:64]: Could you add a test that verifies that `dask config set` does not accidentally nuke unrelated settings in the same file?

File: dask/tests/test_cli.py [hunk 3]
```diff
@@ -34,6 +35,35 @@ def test_config_get_bad_value():
...
+    assert not config_file.exists()
+
+    runner = CliRunner()
+    result = runner.invoke(
+        dask.cli.config_set,
+        ["array.chunk-size", "333MiB"],
+        env={"DASK_CONFIG": str(tmp_conf_dir)},
+    )
+
+    assert config_file.exists()
```
Comment (crusaderky) [inline dask/tests/test_cli.py:59]: Could you actually test file contents? Also test the different cases of
- literal string
- eval'ed complex python object e.g. `[1, 2]`

File: dask/cli.py [hunk 4]
```diff
@@ -75,6 +77,61 @@ def config_get(key=None):
...
+                """Config key not specified. Are you looking for "dask config list"?"""
+            ),
+            err=True,
+        )
+        exit(1)
+    else:
+        try:
+            value = ast.literal_eval(value)
+        except (ValueError, SyntaxError):  # Not a literal
+            pass
```
Comment (crusaderky) [inline dask/cli.py:97]: This is different from the same code in dask/config.py and doesn't process null/none correctly:
https://github.com/dask/dask/blob/9a1d4f14e29f279cd1eb3154367cc0806fcebbe4/dask/config.py#L260-L266

Maybe a minor refactor of `dask.config.collect_env` to parse the value from here would be more robust in the long term than trying to keep the functionality aligned in the two places?

**Comment by milesgranger** [comment]
Thanks for the review @crusaderky, clearly the CLI will take a bit more work, so I've moved the config option over to https://github.com/dask/dask/pull/10925. 

> The new command dumps out your whole dask config to ~/.config/dask/dask.yaml.

Indeed, I wasn't sure if this was correct either but saw this is the same flow being done with `coiled config set` and thus followed suit. And will anyway address the remainder later today.

File: dask/dask.yaml (hunk 1 above)
Comment (milesgranger) [reply dask/dask.yaml]: Renamed and moved to https://github.com/dask/dask/pull/10925

**Review by crusaderky — APPROVED** [review]

**Review by crusaderky — CHANGES_REQUESTED** [review]
Could you document the new command in docs/source/configuration.rst?
Everything else LGTM

**Comment by milesgranger** [comment]
> Could you document the new command in docs/source/configuration.rst?

Ah nuts, sorry I'd forgotten that, done now.

image

**Comment by mrocklin** [comment]
cc @jacobtomlinson @ntabris

**Comment by ntabris** [comment]
this should link to https://github.com/dask/dask/issues/9746 rather than #10917, right?

**Comment by ntabris** [comment]
> > The new command dumps out your whole dask config to ~/.config/dask/dask.yaml.
> 
> Indeed, I wasn't sure if this was correct either but saw this is the same flow being done with `coiled config set`

FYI that's not what `coiled config set` does:

image

**Review by crusaderky — APPROVED** [review]

**Pull request merged**